packages/react/src/index.tsx

```diff
@@ -177,9 +177,14 @@ const WrappedHorizontalLayoutRenderer = <T extends React.ReactNode>(props: {
   const { component: Component, element } = props;
   const { events, type, elements, ...rest } = element;
 
+  const [loaded, setLoaded] = React.useState(false);
+
   React.useEffect(() => {
-    events.onLoad();
-  }, [events]);
+    if (!loaded) {
+      events.onLoad();
```

Comment:
isn't the real issue here that this `useEffect` is being triggered again when it shouldn't? the only way for that to happen is if the `events` object is different; perhaps we're not memoizing it before passing it to this component when we should?

Reply:
I wonder if it's coming from reconstructing the anonymous function here https://github.com/mod-protocol/mod/blob/3b751dba6081cc0864f7c0b68e2c60c152ac5da8/packages/core/src/renderer.ts#L1197

Reply:
not sure - I think perhaps our core renderer doesn't follow react's component model with uniquely deduplicated components (keys) vs just remounting all the components. Could be worth passing a key or trying to debug whether thats what the cause is.

Reply:
did a bit of reading and useEffect compares reference types in the dependency array based on their location in memory. so what i think is happening is once the tree rerenders when the context changes, it creates a new anonymous function which will have a new location in memory and hence retriggers the useEffect hook

when i change the latch to a simple state variable that takes on the initial value of the events prop, it doesn't do the infinite looping anymore:

```typescript
const [events] = React.useState(element.events);

React.useEffect(() => {
  events.onLoad();
}, [events]);
```

updated the pr. lmk what you think